Sources/MessagingPush/MessagingPush+AppDelegate.swift

```diff
@@ -53,60 +53,26 @@ extension MessagingPushImplementation {
         didReceive response: UNNotificationResponse,
         withCompletionHandler completionHandler: @escaping () -> Void
     ) -> Bool {
-        guard let pushContent = userNotificationCenter(center, didReceive: response) else {
-            // push did not come from CIO
-            // Do not call completionHandler() because push did not come from CIO. Another service might have sent it so
-            // allow another SDK
-            // to call the completionHandler()
-            return false
-        }
-
-        switch response.actionIdentifier {
-        case UNNotificationDefaultActionIdentifier: // push notification was touched.
-            if let deepLinkUrl = pushContent.deepLink {
-                // A hack to get an instance of deepLinkUtil without making it a property of the MessagingPushImplementation class. deepLinkUtil is not available to app extensions but MessagingPushImplementation is.
-                // We get around this by getting a instance in this function, only.
-                if let deepLinkUtil = sdkInitializedUtil.postInitializedData?.diGraph.deepLinkUtil {
-                    deepLinkUtil.handleDeepLink(deepLinkUrl)
-                }
-            }
-        default: break
+        // A hack to get an instance of pushClickHandler without making it a property of the MessagingPushImplementation class. pushClickHandler is not available to app extensions but MessagingPushImplementation is.
+        // We get around this by getting a instance in this function, only.
+        if let pushClickHandler = sdkInitializedUtil.postInitializedData?.diGraph.pushClickHandler {
```

Comment:
> This gets called from AppDelegate only right?

Yes. The push click handling swizzling *must* occur during `AppDelegate.didFinishLaunching` function. It's a requirement by the OS to do that. 

> otherwise not sure how reliable would it be for wrappers

You are correct. Currently, the SDK being dependent to app lifecycle is a blocker for this feature to launch on RN and Flutter. The only way to be able to launch this feature on those SDKs is to have customers initialize the iOS SDK in the `AppDelegate` *and* in their dart/js code, which is not ideal. This is not a blocker for Expo because we generate all AppDelegate code so it's easier for us to add the required iOS initialize function as the workaround. 

We have not yet prioritized RN and Flutter for this feature. We may decide to launch it after we make the SDK lifecycle independent. But iOS and Expo, this method works.